static/js/publisher/release/components/defaultTrackModifier.js

```diff
@@ -0,0 +1,220 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import { getTracks, getTrackRevisions } from "../selectors";
+import { openModal, closeModal } from "../actions/modal";
+import { showNotification, hideNotification } from "../actions/notification";
+import { setDefaultTrack } from "../actions/defaultTrack";
+
+class DefaultTrackModifier extends Component {
+  constructor(props) {
+    super(props);
+
+    this.setDefaultTrackHandler = this.setDefaultTrackHandler.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+    this.confirmDefaultTrackHandler = this.confirmDefaultTrackHandler.bind(
+      this
+    );
+    this.clearDefaultTrackHandler = this.clearDefaultTrackHandler.bind(this);
+    this.confirmClearDefaultTrackHandler = this.confirmClearDefaultTrackHandler.bind(
+      this
+    );
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      defaultTrack,
+      currentTrack,
+      closeModal,
+      showNotification,
+      snapName
+    } = this.props;
+
+    if (defaultTrack.track !== prevProps.defaultTrack.track) {
+      closeModal();
+
+      if (defaultTrack.track === null) {
+        showNotification({
+          status: "success",
+          appearance: "positive",
+          content: `The default track for ${snapName} has been removed. All new installations without a specified track (e.g. \`sudo snap install ${snapName}\`) will receive updates from latest track.`,
+          canDismiss: true
+        });
+      } else {
+        showNotification({
+          status: "success",
+          appearance: "positive",
+          content: `The default track for ${snapName} has been set to ${currentTrack}. All new installations without a specified track (e.g. \`sudo snap install ${snapName}\`) will receive updates from the newly defined default track ${currentTrack}.`,
+          canDismiss: true
+        });
+      }
+    }
+  }
+
+  handleCloseModal() {
+    const { closeModal } = this.props;
+
+    closeModal();
+  }
+
+  confirmDefaultTrackHandler() {
+    const { currentTrack, setDefaultTrack, csrfToken, snapName } = this.props;
+
+    setDefaultTrack(snapName, csrfToken, currentTrack);
+  }
+
+  confirmClearDefaultTrackHandler() {
+    const { setDefaultTrack, csrfToken, snapName } = this.props;
+
+    setDefaultTrack(snapName, csrfToken, null);
+  }
+
+  clearDefaultTrackHandler() {
+    const { defaultTrack, openModal } = this.props;
+
+    openModal({
+      title: "Clear default track",
+      content: `By clearing the default track, any device that installed the snap with ${
+        defaultTrack.track
+      } track specified will get updates from the latest track. Would you like to proceed?`,
+      actions: [
+        {
+          className: "p-button--positive u-no-margin--bottom u-float--right",
+          onClick: this.confirmClearDefaultTrackHandler,
```

Comment:
This feels a bit overkill to pass classNames etc through redux state. At least the styling bits should be handled by modal component hopefully.

But most certainly we shouldn't pass functions through redux and keep them as part of the store. Store is meant to be serialisable and immutable at any point, functions are not.

In case of `closeModal` callback, it seems that all we do is calling `closeModal` action - I guess modal can do it itself, it's not this component that needs to handle this?
It may be more tricky in case of more specific handlers like `confirmClearDefaultTrackHandler`. But maybe in their case we could also pass a redux action object (`{ type: "ACTION", payload: { ... }}` rather than function?